lib/TH/THHalf.c

```diff
@@ -0,0 +1,141 @@
+#include "THHalf.h"
+#include "TH.h"
+
+static half half_max = TH_HALF_MAX;
+
+/*
+ * Copyright 1993-2014 NVIDIA Corporation.  All rights reserved.
+ *
+ * NOTICE TO LICENSEE:
+ *
+ * This source code and/or documentation ("Licensed Deliverables") are
```

Comment:
As noted in https://github.com/pytorch/pytorch/issues/654 , the license text that we included here was incorrect.  We are working to get this corrected.